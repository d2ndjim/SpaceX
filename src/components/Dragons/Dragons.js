/* eslint-disable no-unused-vars */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchDragons, updateDragons } from '../../redux/dragons/dragonsSlice';
import styles from './Dragons.module.css';

const Dragons = () => {
  const dragons = useSelector((state) => state.dragons);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchDragons());
  }, [dispatch]);

  const handleClick = (id) => {
    dispatch(updateDragons(id));
  };

  return (
    <section>
      <div className={styles.dragons_wrapper}>
        {
          dragons.map((dragon) => (
            <div key={dragon.id} className={styles.dragon_wrapper}>
              <img src={dragon.image} alt="Dragon" />
              <div className={styles.dragons_info}>
                <h3>{dragon.name}</h3>
                <p>
                  {(dragon.reserved) ? ((<span className={styles.badge}>Reserved</span>)) : ''}
                  {dragon.description}
                </p>
                <button
                  className={(dragon.reserved) ? styles.buttonCancel : styles.buttonDefault}
                  type="button"
                  onClick={() => handleClick(dragon.id)}
                >
                  {(dragon.reserved) ? 'Cancel Reservation' : 'Reserve Dragon'}
                </button>
              </div>
            </div>
          ))
        }
      </div>
    </section>
  );
};

export default Dragons;
