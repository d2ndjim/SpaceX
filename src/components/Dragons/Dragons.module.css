:root {
  --buttonDefault: #007bff;
  --buttonCancel: #6d757d;
  --reserved: #18a2b8;
}

.dragons_wrapper {
  width: 65%;
  margin: 3rem auto;
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

.dragon_wrapper {
  display: flex;
  gap: 1rem;
}

.dragon_wrapper h2 {
  font-size: 1.6rem;
}

.dragons_wrapper img {
  width: 80%;
  height: 80vw;
  max-width: 325px;
  max-height: 300px;
}

button {
  border-radius: 5px;
  font-size: 1.3rem;
  padding: 1rem 1.5rem;
}

.buttonDefault {
  border: 1px solid var(--buttonDefault);
  background-color: var(--buttonDefault);
  color: #fff;
}

.buttonCancel {
  border: 1px solid var(--buttonCancel);
  background-color: #fff;
  color: var(--buttonCancel);
}

.badge {
  background-color: var(--reserved);
  border-radius: 5px;
  color: #fff;
  font-size: small;
  padding: 0.2rem 0.4rem;
  margin: 5px;
}
