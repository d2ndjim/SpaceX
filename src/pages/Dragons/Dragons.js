/* eslint-disable no-unused-vars */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getDragons, selectDragons, reserveDragonsToggle as toggleReserve } from '../../redux/dragons/dragonsSlice';

const Dragons = () => {
  const dragons = useSelector(selectDragons);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getDragons());
  }, [dispatch]);

  return (
    <>
      <div>
        <h1>Hello!</h1>
      </div>
    </>
  );
};

export default Dragons;
