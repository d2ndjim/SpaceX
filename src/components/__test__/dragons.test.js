import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Dragons from '../Dragons/Dragons';

describe('Dragons test', () => {
  it('renders dragons', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Dragons />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
