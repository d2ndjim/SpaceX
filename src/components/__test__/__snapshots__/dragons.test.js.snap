// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragons test renders dragons 1`] = `
<section>
  <div
    className="dragons_wrapper"
  />
</section>
`;
